src/phoenix/trace/dsl/filter.py

```diff
@@ -1,199 +1,440 @@
+        "llm.token_count.total",
+    }
+)
+
+_STRING_NAMES = {
+    "span_id": models.Span.span_id,
+    "trace_id": models.Trace.trace_id,
+    "context.span_id": models.Span.span_id,
+    "context.trace_id": models.Trace.trace_id,
+    "parent_id": models.Span.parent_id,
+    "span_kind": models.Span.span_kind,
+    "name": models.Span.name,
+    "status_code": models.Span.status_code,
+    "status_message": models.Span.status_message,
+}
+_FLOAT_NAMES = {
+    "latency_ms": models.Span.latency_ms,
+    "cumulative_llm_token_count_completion": models.Span.cumulative_llm_token_count_completion,
+    "cumulative_llm_token_count_prompt": models.Span.cumulative_llm_token_count_prompt,
+    "cumulative_llm_token_count_total": models.Span.cumulative_llm_token_count_total,
+}
+# TODO(persistence): find a better home (and a better name) for _NAMES
+_NAMES = {
+    **_STRING_NAMES,
+    **_FLOAT_NAMES,
+    "attributes": models.Span.attributes,
+    "events": models.Span.events,
+}
+_BACKWARD_COMPATIBILITY_REPLACEMENTS = {
+    # for backward-compatibility with the previous implementation
+    "cumulative_token_count.completion": "cumulative_llm_token_count_completion",
+    "cumulative_token_count.prompt": "cumulative_llm_token_count_prompt",
+    "cumulative_token_count.total": "cumulative_llm_token_count_total",
+}
+
+
+# TODO(persistence): remove this protocol
+class SupportsGetSpanEvaluation(typing.Protocol):
+    def get_span_evaluation(self, span_id: SpanID, name: str) -> typing.Optional[pb.Evaluation]: ...
 
 
 @dataclass(frozen=True)
 class SpanFilter:
     condition: str = ""
-    evals: Optional[SupportsGetSpanEvaluation] = None
-    valid_eval_names: Optional[Sequence[str]] = None
+    # TODO(persistence): remove `evals` and `valid_eval_names` from this class
+    evals: typing.Optional[SupportsGetSpanEvaluation] = None
+    valid_eval_names: typing.Optional[typing.Sequence[str]] = None
     translated: ast.Expression = field(init=False, repr=False)
-    compiled: Any = field(init=False, repr=False)
+    compiled: typing.Any = field(init=False, repr=False)
 
     def __bool__(self) -> bool:
         return bool(self.condition)
 
     def __post_init__(self) -> None:
-        condition = self.condition or "True"  # default to no op
-        root = ast.parse(condition, mode="eval")
-        if self.condition:
-            _validate_expression(root, condition, valid_eval_names=self.valid_eval_names)
-        translated = _Translator(condition).visit(root)
+        if not (source := self.condition):
+            return
+        root = ast.parse(source, mode="eval")
+        _validate_expression(root, source, valid_eval_names=self.valid_eval_names)
+        translated = _Translator(source).visit(root)
         ast.fix_missing_locations(translated)
         compiled = compile(translated, filename="", mode="eval")
         object.__setattr__(self, "translated", translated)
         object.__setattr__(self, "compiled", compiled)
```

Comment:
Curious, how robust is this to injections?

Reply:
Should be robust because we are passing the result to sqlalchemy. If sqlalchemy is not robust then this is also not robust.